src/chainstate/stacks/index/node.rs

```diff
@@ -669,9 +669,9 @@ impl fmt::Debug for TrieNode4 {
 }
 
 impl TrieNode4 {
-    pub fn new(path: &Vec<u8>) -> TrieNode4 {
+    pub fn new(path: &[u8]) -> TrieNode4 {
         TrieNode4 {
-            path: path.clone(),
+            path: path.to_owned(),
```

Comment:
Generally speaking in scenarios like this one, I'd argue it's better to take the owned type as input instead of cloning a reference. So I'd change the input to `path: Vec<u8>`, leaving it up to the caller to decide when to clone.

I recognize that this is beyond the scope of the current refactor, so feel free to leave that for future improvements.

Reply:
Yeah, don't do this.

Reply:
@netrome I have fixed where it could be simply fixed. Other places could produce ripples in the code and it should be done in a separate PR.

Reply:
Great, thank you!